File: src/app/components/models/Transaction.ts
import { Compte } from './Compte';
import { TypeTransaction } from './TypeTransaction';

export class Transaction {
  idTransaction?: number;
  montant?: number;
  dateDeTransaction?: Date;
  compteSource?: Compte;
  compteDestinataire!: Compte;
  typeTransaction!: TypeTransaction;
}

File: src/main/java/tn/esprit/spring/Entity/Transaction.java
package tn.esprit.spring.Entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Long idTransaction;
    private double montant;
    private Date dateDeTransaction;
    @ManyToOne
    @JoinColumn(name = "compte_source_id", nullable = false)
    private Compte compteSource;
    @ManyToOne
    @JoinColumn(name = "compte_destinataire_id", nullable = false)
    private Compte compteDestinataire;

    private TypeTransaction typeTransaction;
}
